### DISCUSSION TITLE
500 error on a fresh install

### DISCUSSION THREAD

**Discussion by filip-van-hoeckel** [firefly-iii/firefly-iii#9512] [Firefly III]
Latest Firefly, Ubuntu 20, PHP 8.3, NGINX, on Proxmox (as a LXC, not Docker). Followed the [installation documentation](https://gist.github.com/Engr-AllanG/34e77a08e1482284763fff429cdd92fa
), with a few additional PHP modules when asked by the installation. Opening the app redirects me to the registration page, which leads to a 500 error. Enabled debugging to this error message `Uncaught Illuminate\Foundation\ViteManifestNotFoundException: Vite manifest not found at: /var/www/firefly-iii/public/build/manifest.json in /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Vite.php:934`. 

I have basic *nix knowledge and web development skills. Please help me solve this as I'd love to try the product. Thanks in advance.

**Comment by JC5** [comment]
Those docs seem to be a little out of date. These days the download is from the Releases-tab, and it includes all dependencies. 

See:

https://docs.firefly-iii.org/how-to/firefly-iii/installation/self-managed/

**Comment by filip-van-hoeckel** [comment]
1. Rpi w/ Docker: not my choice
2. instructions I followed, but apparently are outdated
3. retired script, again: using Docker
4. this then?

**Comment by filip-van-hoeckel** [comment]
OK. so [I followed this](https://docs.firefly-iii.org/how-to/firefly-iii/installation/self-managed/) then do 'php artisan firefly-iii:upgrade-database
" then I am greeted with 'Target class [hash] does not exist.'

**Comment by filip-van-hoeckel** [comment]
OK, 'composer dump auto-load' fixed it, apparently.

**Comment by filip-van-hoeckel** [comment]
Spoke too soon, database error occurred. 'SQLSTATE[42S02]: Base table or view not found: 1146 Table 'firefly.invited_users' doesn't exist (Connection: mysql, SQL: select * from `invited_users` where `invite_code` = and `expires` > 2024-12-01 00:00:00 and `redeemed` = 0 limit 1)'

**Reply by JC5** [reply]
`php artisan migrate --seed`?

**Reply by filip-van-hoeckel** [reply]
I trashed the database and just started over as there was no user data present.

**Comment by filip-van-hoeckel** [comment]
OK, redid the installation again and got to the register screen. Registered, unchecked pw security checkbox, only to have it check and deny my password anyway because of length ;-) then was greeted with another 500 error 'An exception has been thrown during the rendering of a template ("Unable to write in the cache directory (/var/www/firefly-iii/storage/framework/views/twig/ae).")'

**Reply by filip-van-hoeckel** [reply]
Found the error in the FAQ and corrected it. I'm in, whew.
